pkg/pgmodel/migrate.go

```diff
@@ -107,22 +107,24 @@ func Migrate(db *pgx.Conn, versionInfo VersionInfo, extOptions extension.Extensi
 // the extension, are set up correctly. This will set the ExtensionIsInstalled
 // flag and thus should only be called once, at initialization.
 func CheckDependencies(db *pgx.Conn, versionInfo VersionInfo, migrationFailedDueToLockError bool, extOptions extension.ExtensionMigrateOptions) (err error) {
-	if err = CheckSchemaVersion(context.Background(), db, versionInfo); err != nil {
+	if err = CheckSchemaVersion(context.Background(), db, versionInfo, true); err != nil { // We pass hasSharedLease as true since it has already been verified at the start.
 		return err
 	}
-
 	return extension.CheckVersions(db, migrationFailedDueToLockError, extOptions)
 }
 
 // CheckSchemaVersion checks the DB schema version without checking the extension
-func CheckSchemaVersion(ctx context.Context, conn *pgx.Conn, versionInfo VersionInfo) error {
+func CheckSchemaVersion(ctx context.Context, conn *pgx.Conn, versionInfo VersionInfo, hasSharedLease bool) error {
```

Comment:
I think there is confusion here what we care about is not the shared lease lock but the exclusive migration lock